### PR TITLE
Add support  for experimental attributes

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

File: experimental/saitypesextensions.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+#ifndef __SAITYPESEXTENSIONS_H_
+#define __SAITYPESEXTENSIONS_H_
+
+#include <saitypes.h>
+
+/**
+ * @brief SAI object type extensions
+ *
+ * @flags
```
Comment (mikelazar) [inline experimental/saitypesextensions.h]: Do we need the flags tag ? Don't we have a contiguous range ?

File: experimental/saitypesextensions.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ */
+typedef enum _sai_object_type_extensions_t
+{
+    SAI_OBJECT_TYPE_EXTENSIONS_RANGE_START = SAI_OBJECT_TYPE_MAX,
+
+    SAI_OBJECT_TYPE_FPGA = SAI_OBJECT_TYPE_EXTENSIONS_RANGE_START,
+
+    /** Add new experimental object types above this line */
```
Comment (mikelazar) [inline experimental/saitypesextensions.h]: You may not want a doxygen comment here ... just a regular comment starting with a /*

File: experimental/saitypesextensions.h (hunk 1 above)
Comment (kcudnik) [reply experimental/saitypesextensions.h]: we have, but @flags is enum bound, and this enum is not starting from "0" but from last enum of corresponding sai object type enum

File: experimental/saitypesextensions.h (hunk 2 above)
Comment (kcudnik) [reply experimental/saitypesextensions.h]: ah yes

**Comment by kcudnik** [comment]
as for empty enum, current constraints on parser don't allow to specify empty enums since they make no sense to the code (excluding start and end range)

File: experimental/saitypesextensions.h (hunk 1 above)
Comment (mikelazar) [reply experimental/saitypesextensions.h]: OK, thanks

**Comment by kcudnik** [comment]
test this please

**Pull request merged**